### DISCUSSION TITLE
Bitclout Themes and Minor UX Changes/Adjustments— Should each be a CIP?

### DISCUSSION THREAD

**Discussion by omicreativedev** [deso-protocol/dips#44] [Q&A]
Based on the present CIP model, themes and minor UX changes not requiring a package were in the Small task bucket.

I authored cake-theme.ts and others have contributed several recently, as well as, several cool user interface improvements.

I'm interested in community consensus about the weight of significance of theme additions and changes to user interface.

Should these have a bucket? Are they a big deal?

Some folks brought up a good question in my recent pull for book theme here: https://github.com/bitclout/frontend/pull/230

How many themes do we want? How many do we need? And what kind of themes?

Also note: Bitflare now allows publishing front end nodes in a click! I don't imagine everyone doing so will know how to theme but MAY want to have a different look on their frontend to start. Likewise, for new node developers before they get into the code and really make something special. So it's my opinion it would be well to preload those we know people would use it for (ie. communities, guilds, development, social networks, etc.)

Likewise, I brought up an issue with inconsistent shape of avatars in the Issues of the frontend repository.

That begs the question, what do WE want Bitclout.com MAIN to look like? And how many options do we want to give users for browsing experience? Or frontend owners? Or new independent node operators?

I would love to know what everyone thinks about workflow, ideas, etc. 

Most specifically I'd like to know how people want the LOOK and AESTHETIC and VIBE of the site to resonate with the userbase.

- Should each theme become a CIP?
- If yes, what level of CIP?
- At what point should UX changes/fixes become a CIP?
- How many themes should Bitclout have on main node?
- If limited number of themes, what themes do we want/need by default?
- What would you like to see the LOOK of Bitclout.com base (for all other nodes by default) become- not theme but design (round/square/rounded avatars? — thin/thick borders? — animated/not? — fancy javastuff/not? etc)

My approach was thinking that people who want to build on Bitclout will set up a node of their own and want to have a great user platform out the box with a variety of options to look customized and different enough before they go neck deep in making it truly their own. Some people might just want a cool home page.

I would LOVE any and all open discussion on the topic.

**Comment by lazynina** [comment]
1. I don't think each theme should be a CIP - that would be overkill.
2. n/a
3. Anything that redesigns a component is considered medium, so significant changes to the interface would fall in that category. Smaller things would be a small change.
4. This is what I think we need to discuss. Adding a lot of themes adds a lot of overhead to ensure each theme is appropriately supported with each new UI change.  I would only like to support themes that have a strong use case. Additionally, it's really pretty low effort to add a theme to a third party node and think it would be something great for node operators to make themselves unique. Perhaps we create a separate repo to manage these themes and node operators can import which ones make sense for their node, even if it's not on bitclout.com

**Reply by brootle** [reply]
Please support my low contrast dark theme! Thank you! 😀

**Reply by omicreativedev** [reply]
I have already done just that. Great minds think alike! I have a repo for themes and tweaks for nodes, open-source Bitclout plugin hacks, etc. Not much in it yet, but that was the plan to have a bunch of "stuff" that isn't going to come boxed with the program.

I think it would be awesome to have an OFFICIAL theme-bin too! It could be noted in the Contributing MD that contributions MUST note the latest version of frontend that the theme is tested compatible with. And that contributors are responsible for updating and testing themes regularly. And that themes that are left incompatible for 30 days will get purged or updated by the community. Then the only overhead is throwing out the trash once per month.

As for on MAIN, yeah I understand. Updating even 20 themes is extra work. I think there is valid need and desire for certain color schemes though but again, up to yalls discretion. I would totally help with a themebin repo.

**Comment by tijno** [comment]
What are the critical themes that are missing?

Should there be a super high contrast one ?
What about a theme best suited for visually impaired? 

Not my area of expertise - but probably should have at least 1 specifically helping those ?
